Make headers usable alone, include the headers they need.
There are probably much more missing.

--- sys/net/if.h.old	2009-05-15 10:51:50 +0000
+++ sys/net/if.h	2009-05-15 10:52:05 +0000
@@ -34,6 +34,7 @@
 #define	_NET_IF_H_
 
 #include <sys/cdefs.h>
+#include <sys/socket.h>
 
 #ifdef _KERNEL
 #include <sys/queue.h>
